numba/targets/builtins.py

```diff
@@ -25,6 +25,14 @@ def generic_is_not(context, builder, sig, args):
     return builder.not_(is_impl(builder, args))
 
 
+@lower_builtin('is', types.Any, types.Any)
+def generic_is(context, builder, sig, args):
+    """
+    Default implementation for `x is y`
+    """
+    return cgutils.false_bit
```

Comment:
I think this is wrong and should piggyback on the `==` operator. Otherwise, trivial code such as `x is x` will return False unexpectedly.